File: Park-Dasol/64061/programmers64061.py
def solution(board, moves):
    answer = 0
    N = len(board)
    height = [0] *N
    for j in range(N):
        for i in range(N):
            if board[i][j] != 0:
                height[j] = i
                break
    s = []
    for m in moves:
        move = m -1
        if height[move] < N:
            temp = board[height[move]][move]
            if len(s) and s[-1] == temp:
                s.pop(-1)
                answer += 2
            else :
                s.append(temp)
            board[height[move]][move] = 0
            height[move] += 1


    print(answer)
    return answer

solution([[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]], [1,5,3,5,1,2,1,4])

File: HanJaehee/programmers/64061/programmers64061.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int picked, idx, answer = 0;
		Stack<Integer> s = new Stack<Integer>();
		int[] tmp;

		for (int cmd : moves) {
			cmd--;
			tmp = rtoc(board, cmd);
			idx = getDoll(tmp);
			if (idx == -1)
				continue;
			picked = board[idx][cmd];
			board[idx][cmd] = 0;
			if (!s.isEmpty() && s.peek() == picked) {
				answer += 2;
				s.pop();
				continue;
			}
			s.add(picked);
		}
		return answer;
    }
    
    private int[] rtoc(int[][] board, int num) {
		int[] arr = new int[board.length];
		for (int j = 0; j < board.length; j++) {
			arr[j] = board[j][num];
		}

		return arr;
	}

	private int getDoll(int[] arr) {
		for (int i = 0; i < arr.length; i++) {
			if (arr[i] != 0) {
				return i;
			}
		}
		return -1;
	}
}